### DISCUSSION TITLE
Automatically test all packages if no changes occur

### DISCUSSION THREAD

**Discussion by redonkulus** [lerna/lerna#3604] [Q&A]
I have a use case where I run `lerna run test —since` for only packages that changed since the previous git tag. This works great as of today. 

However, if no packages have changed, instead of doing nothing, I’d like to execute all package tests. This is useful in scenarios where a root package has been updated and you want to know if it breaks any of the packages (e.g., jest upgrade). 

Is there a way to do this in lerna today? If not, is there a work around?

**Comment by fahslaj** [comment]
I think your best option would be to use `lerna changed` to get a list of packages that have changed since the most recent tag. If any packages have changed, then run `lerna run test --since` as usual. If `lerna changed` reports that no packages have changed, then you can run `lerna run test`, which will include all packages.

**Reply by redonkulus** [reply]
Would you just wire this up in npm run scripts? Or would you write a custom script?

**Reply by fahslaj** [reply]
I would probably start with a npm script, but you could extract it to a custom script that is run via an npm script if that makes it easier.

**Reply by redonkulus** [reply]
I ended up with this:

```bash
lerna changed --json | wc -l | xargs test 0 -eq && lerna run lint || lerna run lint --since
```

Its a bit verbose, would be nice if lerna had a way to do automatically. Maybe something like:

```bash
lerna run lint --since --all
```

`--all` is a bad flag name but something that would tell lerna to execute all if no packages have changed.

@JamesHenry for his thoughts.
